src/VisualStudio/Core/Def/Implementation/Library/FindResults/TreeItems/MetadataDefinitionTreeItem.cs

```diff
@@ -36,7 +35,17 @@ public MetadataDefinitionTreeItem(ISymbol definition, ushort glyphIndex)
 
         public override int GoToSource()
         {
-            return VSConstants.E_NOTIMPL;
+            var resolution = _symbolKey.Resolve(_workspace.CurrentSolution.GetCompilationAsync(_referencingProjectId, CancellationToken.None).Result);
```

Comment:
Should we be popping up a threaded wait dialog then? Or is this a programmatic access point that wouldn't be happy with that?